pkg/cloudconformity/client.go

```diff
@@ -7,31 +7,54 @@ import (
 	"time"
 )
 
+var (
+	V1HostMap = map[string]string{
+		"au":    "api.au.xdr.trendmicro.com",    // Australia
+		"eu":    "api.eu.xdr.trendmicro.com",    // Europe
+		"in":    "api.in.xdr.trendmicro.com",    // India
+		"jp":    "api.xdr.trendmicro.co.jp",     // Japan
+		"sg":    "api.sg.xdr.trendmicro.com",    // Singapore
+		"mea":   "api.mea.xdr.trendmicro.com",   // United Arab Emirates
+		"us":    "api.xdr.trendmicro.com",       // US
+		"usgov": "api.usgov.xdr.trendmicro.com", // US Gov
+	}
+)
+
 type Client struct {
-	Region     string
-	Apikey     string
-	Url        string
-	HttpClient *http.Client
+	Region       string
+	Apikey       string
+	Url          string
+	HttpClient   *http.Client
+	UseV1Feature bool
 }
 
 // create a client with region and apiKey
-func NewClient(region string, apikey string) (*Client, error) {
-	client := Client{Region: region, Apikey: apikey, Url: getUrl(region), HttpClient: &http.Client{
+func NewClient(region string, apikey string, useV1Feature bool) (*Client, error) {
+	Url, err := getUrl(region, useV1Feature)
+	if err != nil {
+		return nil, err
+	}
+
+	client := Client{Region: region, Apikey: apikey, UseV1Feature: useV1Feature, Url: Url, HttpClient: &http.Client{
 		Timeout: time.Second * 30,
 	}}
 
-	_, err := client.validateApiKey()
-	if err != nil {
-		return nil, err
+	// only validate API key if not using v1 feature
+	if !useV1Feature {
+		_, err = client.validateApiKey()
```

Comment:
I understand V1 doesn't offer an API to validate keys.
Instead of skipping validation completely, could we use one of the GET APIs with the key to check? E.g. if a 4xx error is returned, they key is invalid